src/geopathfinder/folder_naming.py

```diff
@@ -845,10 +845,63 @@ def remove_smartpath(self, key):
         self.count_dirs()
 
 
-    def trim2branch(self, level, pattern, register_file_pattern=None):
+    def get_subtree_matching(self, level, pattern, register_file_pattern=None):
         '''
-        Returns a branch (a subtree) of a SmartTree that matches with
-        the pattern at the given level.
+        Returns a subtree of the SmartTree with branches comprising
+        ALL matches with the pattern at the given level.
+
+        Parameters
+        ----------
+        level : str
+            Name of level in hierarchy.
+            e.g. 'wflow'
+        pattern : str
+            string defining search pattern at given level
+            e.g. 'C1003'
+
+        Returns
+        -------
+        branch : SmartTree
+            SmartTree object that describes the seeked branch,
+            part of current SmartTree
+        '''
+
+        branch = copy.deepcopy(self)
+
+        branch_paths = self.collect_level_string(level, pattern=pattern, unique=True)
+
+        if len(branch_paths) == 0:
+            warnings.warn('get_subtree_unique_rebased(): No matches for "pattern" at "level"!')
+            return NullSmartTree(self.root)
+        else:
+            for d in self.dirs.keys():
+                match = [x for x in branch_paths if x in d]
+                if len(match) == 0:
+                    branch.remove_smartpath(d)
+
+            # update stats
+            branch.count_dirs()
+            file_register = []
+            file_count = 0
+            # update file_register
+            if register_file_pattern is not None:
+                for sp in branch.dirs.values():
+                    sp.build_file_register(pattern=register_file_pattern)
+                    file_register += sp.file_register
+                    file_count += sp.file_count
+
+            branch.file_register = file_register
+            branch.file_count = file_count
+            branch.has_register = True
+
+            return branch
+
+
+    def get_subtree_unique_rebased(self, level, pattern, register_file_pattern=None):
```

Comment:
Confusing function, why combining rebasing and and unique? You could call `get_subtree_matching(...).rebase(level)` - I think this would be more pythonic.

Reply:
yeah, well, the logic is different.
get_subtree_matching() returns a subtree with all branches matching
get_subtree_unique_rebased() returns a single branch, rebased/re-rooted to the given level.

Happy for suggestions about the naming

Reply:
I agree